image_data_guide/02_structuring_data.ipynb

```diff
@@ -0,0 +1,198 @@
+    "    \n",
+    "with open('pickled_data/category_labels.pickle', 'rb') as f:\n",
+    "    category_labels = pickle.load(f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Divide the data into train, validation and test splits\n",
+    "We should divide our data into train, validation and test splits. A typical split ratio is 80/10/10. Our image classification algorithm will train on the first 80% (training) and evaluate its performance at each epoch with the next 10% (validation) and we'll give our model's final accuracy results using the last 10% (test). It's important that before we split the data we make sure to shuffle it randomly so that class distribution among splits is roughly proportional."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.random.seed(0)\n",
+    "image_ids = sorted(list(sample_annos.keys()))\n",
+    "np.random.shuffle(image_ids)\n",
+    "first_80 = int(len(image_ids) * 0.8)\n",
+    "next_10 = int(len(image_ids) * 0.9)\n",
+    "train_ids, val_ids, test_ids = np.split(image_ids, [first_80, next_10])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Make new folder structure and copy image files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 2200/2200 [00:00<00:00, 3945.20it/s]\n",
+      "100%|██████████| 275/275 [00:00<00:00, 6540.22it/s]\n",
+      "100%|██████████| 275/275 [00:00<00:00, 6157.39it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "unstruct_dir = Path('data_sample_2750')\n",
+    "struct_dir = Path('data_structured')\n",
+    "struct_dir.mkdir(exist_ok=True, parents=True)\n",
+    "\n",
+    "for name, split in zip(['train', 'val', 'test'], [train_ids, val_ids, test_ids]):\n",
+    "    split_dir = struct_dir / name\n",
+    "    split_dir.mkdir(exist_ok=True)\n",
+    "    for image_id in tqdm(split):\n",
+    "        category_dir = split_dir / f'{category_labels[sample_annos[image_id][\"category_id\"]]}'\n",
+    "        category_dir.mkdir(exist_ok=True)\n",
+    "        source_path = (unstruct_dir / sample_annos[image_id]['file_name']).as_posix()\n",
+    "        target_path = (category_dir / sample_annos[image_id]['file_name']).as_posix()\n",
+    "        shutil.copy(source_path, target_path)"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "conda_python3",
+   "language": "python",
+   "name": "conda_python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.6.10"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 4
+}
```

Comment:
End notebook by explaining what is next in the tutorial series?

Reply:
Added next steps section.